### PR TITLE
8 extend plankton for docker it

### PR DESCRIPTION
**Pull request by ChristianLohmann**

### CODE REVIEW

File: job.js [hunk 1]
```diff
@@ -34,10 +37,9 @@ function successful(cmd) {
 
 function dockerLogin() {
     return new Promise((resolve, reject) => {
-        // shelljs can't do expansion of ~ :(
         var paths = IN_PROD ?
                     ['/root/.docker/config.json'] :
-                    ['/Users/npiccolotto/.docker/config.json'];
+                    [os.homedir() + '/.docker/config.json'];
```
Comment (prayerslayer) [inline job.js]: Ah, nice :+1:

File: job.js [hunk 2]
```diff
@@ -164,6 +167,9 @@ function job() {
     winston.info('==== new run ====');
     var image = createRandomImage();
     dockerLogin()
+    .catch(output => {
```
Comment (prayerslayer) [inline job.js]: `.catch(winston.error.bind(winston))`

Comment (ChristianLohmann) [inline job.js]: ah, ok. what exactly happens when binding winston to one of its members?

**Comment by prayerslayer** [comment]
I find this approach a bit redundant because `DOCKER_TEST_STATUS` is a function of `LAST_PUSH` and `LAST_PULL`. So the only thing it really adds is a progress indicator. You could introduce a boolean `TEST_IN_PROGRESS` and in the `dockerTestResult` function calculate the status code from `TEST_IN_PROGRESS`, `LAST_PUSH` and `LAST_PULL`.

File: job.js (hunk 1 above)
Comment (ChristianLohmann) [inline job.js]: maybe we could simplify this by omitting the root home dir at all. `os.homedir()` should be intelligent enough to recognize that as well.

File: job.js (hunk 2 above)
Comment (prayerslayer) [inline job.js]: `fn.bind` gives you `fn` with some parameters already applied ("partial application"). In JS the [first parameter to bind](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind) has to be a reference to what `this` inside of `fn` should be.

So this line gives us `winston.error` awaiting stuff it should log to stderr.

**Comment by ChristianLohmann** [comment]
hm ok, i just have seen that there is already some success property in the output map of LAST_PULL and LAST_PUSH. Then you're right and it makes sense to consolidate it.

File: job.js [hunk 3]
```diff
@@ -6,26 +6,29 @@ var shell = require('shelljs'),
...
     LAST_PULL = {},
     LAST_PUSH = {},
+    DOCKER_MISSING_OR_BROKEN = false,
     IN_PROD = process.env.NODE_ENV === 'production',
     JOB_INTERVAL = 1000 * (parseInt(process.env.JOB_INTERVAL, 10) || 10),
-    DOCKER_REGISTRY =  IN_PROD ?
-                        process.env.DOCKER_REGISTRY :
-                        process.env.DEFAULT_DOCKER_REGISTRY,
+    DOCKER_REGISTRY = IN_PROD ?
+        process.env.DOCKER_REGISTRY :
```
Comment (prayerslayer) [inline job.js]: 😐

Comment (ChristianLohmann) [inline job.js]: sorry, this was the default intellij formatter for js

**Pull request merged**